src/Core/Services/MetadataProviders/Converters/ObjectConverter.cs

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Azure.DataApiBuilder.Core.Services.MetadataProviders.Converters
+{
+    /// <summary>
+    /// This is a converter to serialize and deserialize object type
+    /// can object can be of different types int, long, string, datetime. this converter handles different basic object types supported by
+    /// Hotchocolate
+    /// </summary>
+    public class ObjectConverter : JsonConverter<object>
+    {
+        public override object Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            switch (reader.TokenType)
+            {
+                case JsonTokenType.Number:
+                    if (reader.TryGetInt32(out int intValue))
+                    {
+                        return intValue;
+                    }
+                    else if (reader.TryGetInt64(out long longValue))
+                    {
+                        return longValue;
+                    }
+                    else if (reader.TryGetDecimal(out decimal decimalValue))
+                    {
+                        return decimalValue;
+                    }
+                    else if (reader.TryGetSingle(out float floatValue))
+                    {
+                        return floatValue;
+                    }
+                    else if (reader.TryGetDouble(out double doubleValue))
+                    {
+                        return doubleValue;
+                    }
+                    else
+                    {
+                        throw new JsonException($"Unable to deserialize number value. Token: {reader.TokenType}");
+                    }
```

Comment:
Seems like order matters here, right?  because you'd have to attempt each and every conversion if, let's say, you were trying to parse out a `double`

Reply:
I have updated the implementation of object converter itself, to increase the reliability, rather than  maintaining the switch for each case. I am leveraging similar approach as that of the DatabaseObjectOnverter to add extra property for Type